### PR TITLE
[react] deprecate more factory-related types

### PR DESCRIPTION
**Pull request by henryqdineen**
I noticed that `ClassicFactory` a bunch of other types werenot deprecated but is removed in the v19 types. This PR simply deprecates these in v18 types. 

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

### CODE REVIEW

**Review by eps1lon — COMMENTED** [review]
What about the other types related to Factories?

**Comment by henryqdineen** [comment]
> What about the other types related to Factories?

Good point. Originally I was patching our types to remove `createClass` based stuff and noticed that it exposed some unused types. It definitely makes more sense to expand this PR to deprecate the other stuff. Pushed some changes in 60855bb633aad5ee59e464cce994ef7866f9b862

**Review by eps1lon — APPROVED** [review]

File: types/react/v18/ts5.0/index.d.ts
Comment (eps1lon) [inline types/react/v18/ts5.0/index.d.ts:1]: This is missing some deprecations that `v18/index.d.ts` has

**Review by eps1lon — CHANGES_REQUESTED** [review]

File: types/react/v18/ts5.0/index.d.ts
Comment (henryqdineen) [reply types/react/v18/ts5.0/index.d.ts:1]: Thanks. Took care of that in a7664279f0703df79d5fad633d296c61023d1a14

**Review by eps1lon — APPROVED** [review]
Thank you!

**Pull request merged**